crates/aranya-client-capi/src/api/defs.rs

```diff
@@ -1350,69 +1361,445 @@ pub unsafe fn aqc_remove_net_identifier(
+/// @param label_id the AQC channel label ID [`LabelId`] to create the channel with.
+/// @param channel the AQC channel object [`AqcSenderChannel`].
+///
+/// @relates AranyaClient.
+pub unsafe fn aqc_create_uni_channel(
+    client: &mut Client,
+    team: &TeamId,
+    peer: NetIdentifier,
+    label_id: &LabelId,
+    channel: &mut MaybeUninit<AqcSenderChannel>,
+) -> Result<(), imp::Error> {
     // SAFETY: Caller must ensure `peer` is a valid C String.
     let peer = unsafe { peer.as_underlying() }?;
-    let chan_id = client.rt.block_on(client.inner.aqc().create_bidi_channel(
+
+    let client = client.deref_mut();
+    let chan = client.rt.block_on(client.inner.aqc().create_uni_channel(
         team.into(),
         peer,
         label_id.into(),
     ))?;
-    Ok(AqcBidiChannelId {
-        id: chan_id.aqc_id().into_id().into(),
-    })
+
+    Safe::init(channel, imp::AqcSenderChannel::new(chan));
+    Ok(())
 }
 
+/* TODO(nikki): these will panic currently, so let's stub them
+
 /// Delete a bidirectional AQC channel.
 ///
 /// @param client the Aranya Client [`Client`].
-/// @param chan the AQC channel ID [`AqcBidiChannelId`] of the channel to delete.
+/// @param channel_id the AQC Channel [`AqcBidiChannel`] to delete.
 ///
 /// @relates AranyaClient.
 pub fn aqc_delete_bidi_channel(
     client: &mut Client,
-    chan: &AqcBidiChannelId,
+    channel: &AqcBidiChannelId,
 ) -> Result<(), imp::Error> {
+    // TODO(nikki): change to AqcBidiChannel
     let client = client.deref_mut();
     client
         .rt
-        .block_on(client.inner.aqc().delete_bidi_channel(chan.into()))?;
+        .block_on(client.inner.aqc().delete_bidi_channel(channel.into()))?;
     Ok(())
 }
 
 /// Delete a unidirectional AQC channel.
 ///
 /// @param client the Aranya Client [`Client`].
-/// @param chan the AQC channel ID [`AqcUniChannelId`] of the channel to delete.
+/// @param chan the AQC Channel [`AqcSenderChannel`] to delete.
 ///
 /// @relates AranyaClient.
 pub fn aqc_delete_uni_channel(
     client: &mut Client,
-    chan: &AqcUniChannelId,
+    channel: &AqcSenderChannel,
 ) -> Result<(), imp::Error> {
     let client = client.deref_mut();
     client
         .rt
-        .block_on(client.inner.aqc().delete_uni_channel(chan.into()))?;
+        .block_on(client.inner.aqc().delete_uni_channel(channel.into()))?;
+    Ok(())
+}
+*/
+
+/// Waits until an AQC channel is received from the client.
+///
+/// Returns `ARANYA_ERROR_AQC_SERVER_CLOSED` if the server connection was closed.
+///
+/// @param client the Aranya Client [`Client`].
+/// @param channel the AQC channel holder [`AqcChannel`].
+///
+/// @relates AranyaClient.
+pub fn aqc_receive_channel(
+    client: &mut Client,
+    channel: &mut MaybeUninit<AqcChannel>,
+) -> Result<(), imp::Error> {
+    let client = client.deref_mut();
+    let chan = client
+        .rt
+        .block_on(client.inner.aqc().receive_channel())
+        .ok_or(imp::Error::AqcServerClosed)?;
```

Comment:
This is a non-fatal error. If a channel is not received, the caller can try again.
```suggestion
        .ok_or(imp::Error::AqcChannelNotReceived)?;
```

Reply:
If it returns None, it means the channel is closed, AqcClient::receive_channel clearly states that (and so does mpsc::channel)